libr/bin/format/dmp/dmp64.c

```diff
@@ -0,0 +1,236 @@
+/* radare2 - LGPL - Copyright 2020 - abcSup */
+
+#include <r_types.h>
+#include <r_util.h>
+
+#include "dmp64.h"
+
+static int r_bin_dmp64_init_memory_runs(struct r_bin_dmp64_obj_t *obj) {
+	int i, j;
+	dmp64_p_memory_desc *mem_desc = &obj->header->PhysicalMemoryBlockBuffer;
+	if (!memcmp (mem_desc, DMP_UNUSED_MAGIC, 4)) {
+		eprintf ("Warning: Invalid PhysicalMemoryDescriptor\n");
+		return false;
+	}
+	ut64 num_runs = mem_desc->NumberOfRuns;
+	if (num_runs * sizeof (dmp_p_memory_run) >= 0x348) {
+		eprintf ("Warning: Invalid PhysicalMemoryDescriptor\n");
+		return false;
+	}
+	obj->pages = r_list_newf (free);
+	if (!obj->pages) {
+		return false;
+	}
+	dmp_p_memory_run *runs = calloc (num_runs, sizeof (dmp_p_memory_run));
+	if (r_buf_read_at (obj->b, 0x98, (ut8*)runs, num_runs * sizeof (dmp_p_memory_run)) < 0) {
```

Comment:
this other magic number

Reply:
Offset to NumberOfPages property in relative to the header base (0x0).

```
0x88  : PhysicalMemoryBlockBuffer      ['_PHYSICAL_MEMORY_DESCRIPTOR']
```
```
typedef struct _PHYSICAL_MEMORY_DESCRIPTOR64 {
	ut32 NumberOfRuns; // 0x0
	ut32 _padding1;
	ut64 NumberOfPages; // 0x8
	dmp_p_memory_run Run[1];
} dmp64_p_memory_desc;
```